docs/04-api-reference/front-commerce-core/graphql.mdx

```diff
@@ -10,43 +10,80 @@ description:
+| `loadRuntime`      | `function` | The import that will load [GraphQL runtime](#creategraphqlruntime) |
+| `typeDefs`         | `string`   | The GraphQL custom type definitions                                |
+| `schemaDirectives` | `object`   | The GraphQL custom directives                                      |
+
+### Example
 
 ```ts
 import { createGraphQLModule } from "@front-commerce/core/graphql";
-import AcmeModule from "./acmeModule";
+import fooDirective from "./directives/fooDirective";
 
 export default createGraphQLModule({
-  namespace: "Acme",
-  modules: [AcmeModule],
+  namespace: "MyModule",
+  dependencies: ["Front-Commerce/Payment"],
+  loadRuntime: () => import("./runtime"),
+  typeDefs: /* GraphQL */ `
+    type Query {
+      getFoo: Foo
+    }
+    type Foo {
+      id: ID!
+      bar: Int
+    }
+  `,
+  schemaDirectives: {
+    fooDirective,
+  },
 });
 ```
 
-| Argument    | Type                   | Description                         |
-| ----------- | ---------------------- | ----------------------------------- |
-| `namespace` | `string`               | The namespace of the GraphQL module |
-| `modules`   | `Array<GraphQLModule>` | The modules to inject in namespace  |
+## `createGraphQLRuntime`
 
-## `readGraphQLModules`
+Create a GraphQL runtime.
 
-Reads and parses GraphQLModules generated by codegen.
+### Parameters
 
-_Example_
+| Name              | Type                | Description                             |
+| ----------------- | ------------------- | --------------------------------------- |
+| `resolvers`       | `object`            | GraphQL resolvers                       |
+| `contextEnhancer` | `function(context)` | function to enhance the GraphQL context |
 
-```ts
-import { readGraphQLModules } from "@front-commerce/core/graphql";
+### `contextEnhancer` function
 
-const generatedSchemaModules = await readGraphQLModules();
-```
+The context enhancer function recieves the following arguments
 
-## `schema`
+| Argument   | Type      | Description                                                       |
+| ---------- | --------- | ----------------------------------------------------------------- |
+| `config`   | `Config`  | The [application configuration](/docs/3.x/category/configuration) |
+| `user`     | `User`    | The actual user session                                           |
+| `services` | `Service` | The registered services                                           |
+| `loader`   | `Loader`  | Loaders registred in your application                             |
 
-Reads the generated GraphQL schema.
+### Example
 
 ```ts
-import { schema } from "@front-commerce/core/graphql";
-
-const acmeSchema = await schema();
+import { createGraphQLRuntime } from "@front-commerce/core/graphql";
+import createFooContextEnhancer from "./createFooContextEnhancer";
+import resolvers from "./resolvers";
+
+export default createGraphQLRuntime({
+  resolvers: resolvers,
```

Comment:
We are aiming at inlining resolvers for the future, it would be good to lead by example, so instead of importing here we could add a simple resolver which uses Foo?

Reply:
Added inline resolver and a inline typeDefs, let me know your what you think